mathesar/database/collections.py

```diff
@@ -0,0 +1,40 @@
+from sqlalchemy import Column, Integer, String, Table
+
+from mathesar.database.applications import DBApplication
+from mathesar.database.base import ID, db_name, engine, metadata
+
+
+class DBCollection(Table):
+    DEFAULT_COLUMNS = [
+        Column(ID, Integer, primary_key=True),
+    ]
+
+    def insert_rows(self, rows):
+        with engine.begin() as connection:
+            result = connection.execute(self.insert(), rows)
+            return result
+
+    @classmethod
+    def create_from_columns(cls, name, column_names):
+        """
+        This method creates a Postgres table corresponding to the collection.
+        """
+        db_application = DBApplication(name)
+        db_application.create()
```

Comment:
The logic for whether a schema needs to be created lives in the `DBApplication` create method currently. I'll think about a way to make it more explicit.

I figured we would make the application name configurable later, but you're right, it would be better to do it now.